src/agent/onefuzz-result/src/job_result.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use anyhow::Result;
+use async_trait::async_trait;
+use reqwest::Url;
+use serde::{self, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::{
+    collections::HashSet,
+    sync::{Arc, Mutex},
+};
+use storage_queue::QueueClient;
+use tokio::sync::Notify;
+use uuid::Uuid;
+
+#[derive(Debug, Deserialize, Serialize, Hash, Eq, PartialEq, Clone)]
+#[serde(tag = "type")]
+pub enum JobResultData {
+    NewCrashingInput,
+    NoReproCrashingInput,
+    NewReport,
+    NewUniqueReport,
+    NewRegressionReport,
+    NewCoverage,
+    CoverageData,
+    RuntimeStats,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct JobResult {
+    task_id: Uuid,
+    job_id: Uuid,
+    machine_id: Uuid,
+    machine_name: String,
+    data: Vec<JobResultData>,
```

Comment:
I'm not sure I understand why `data` is a `Vec` here. We only expose it through `send_direct` which takes a single `JobResultData`.

Reply:
Leftover from a previous design. I based this off the heartbeat which can send messages with 'cumulative' events. Agreed that it should be removed here.